*** /usr/src/new/m11/Makefile.old	Sun Jan 28 17:47:01 1996
--- /usr/src/new/m11/Makefile	Tue Nov 20 13:06:19 2001
***************
*** 1,4 ****
! # m11 Makefile.  Revised 1996/1/28
  #
  # Note that the pathname for macxrf is wired into exec.m11 ...
  
--- 1,4 ----
! # m11 Makefile.  Revised 2001/11/20
  #
  # Note that the pathname for macxrf is wired into exec.m11 ...
  
***************
*** 5,11 ****
  DESTDIR=
  
  M11=	./m11.x
- L11=	./l11.x
  
  SEPFLAG=-i
  MFLAGS=
--- 5,10 ----
***************
*** 30,36 ****
  all: m11 macxrf m11.0
  
  m11: ${OBJS}
! 	${L11} -ls ${SEPFLAG} ${OBJS}
  	mv exec.out m11
  
  macxrf: macxrf.c
--- 29,35 ----
  all: m11 macxrf m11.0
  
  m11: ${OBJS}
! 	l11 -ls ${SEPFLAG} ${OBJS}
  	mv exec.out m11
  
  macxrf: macxrf.c
*** /usr/src/new/m11/exec.m11.old	Sat Jun 20 13:33:51 1987
--- /usr/src/new/m11/exec.m11	Mon Nov 19 22:26:30 2001
***************
*** 133,139 ****
  errrol=	1
  	.mcall	(at)param
  
! 	.globl	$creat, $open, $close, $exit, $read, $write, $sbrk
  	.globl	$seek, $gettod, $fork, $wait, $execv
  
  				;init sectors
--- 133,139 ----
  errrol=	1
  	.mcall	(at)param
  
! 	.globl	$open, $close, $exit, $read, $write, $sbrk
  	.globl	$seek, $gettod, $fork, $wait, $execv
  
  				;init sectors
***************
*** 648,658 ****
  11$:	call	src.ap		;set up name in linbuf
  	mov	#linbuf,$crtnm	;  and pointer to name
  2$:	mov	$crtmd,-(sp)
  	mov	$crtnm,-(sp)
  	tst	-(sp)
! 	$creat
  	bcs	99$
! 	add	#6.,sp
  	br	7$
  99$:
  	add	#6.,sp
--- 648,659 ----
  11$:	call	src.ap		;set up name in linbuf
  	mov	#linbuf,$crtnm	;  and pointer to name
  2$:	mov	$crtmd,-(sp)
+ 	mov	#3001,-(sp)	; O_CREAT | O_TRUNC | O_WRONLY
  	mov	$crtnm,-(sp)
  	tst	-(sp)
! 	$open
  	bcs	99$
! 	add	#8.,sp
  	br	7$
  99$:
  	add	#6.,sp
*** /usr/src/new/m11/syscalls.m11.old	Fri Feb 27 07:31:11 1987
--- /usr/src/new/m11/syscalls.m11	Mon Nov 19 22:25:23 2001
***************
*** 1,4 ****
! 	.globl	$creat, $open, $close, $exit, $read, $write, $sbrk
  	.globl	$seek, $gettod, $fork, $wait, $execv
  
  	$exit	= ^O104401
--- 1,4 ----
! 	.globl	$open, $close, $exit, $read, $write, $sbrk
  	.globl	$seek, $gettod, $fork, $wait, $execv
  
  	$exit	= ^O104401
***************
*** 8,14 ****
  	$open	= ^O104405
  	$close	= ^O104406
  	$wait	= ^O104524
- 	$creat	= ^O104410
  	$execv	= ^O104413
  	$gettod	= ^O104564
  	$sbrk	= ^O104505
--- 8,13 ----
*** /VERSION.old	Sat Oct 13 17:42:00 2001
--- /VERSION	Mon Nov 19 20:14:53 2001
***************
*** 1,5 ****
! Current Patch Level: 439
! Date: October 13, 2001
  
  2.11 BSD
  ============
--- 1,5 ----
! Current Patch Level: 440
! Date: November 19, 2001
  
  2.11 BSD
  ============
